runtime/crypto_test.go

```diff
@@ -648,3 +650,140 @@ func TestAggregateBLSPublicKeys(t *testing.T) {
+		"f851808080a009833150c367df138f1538689984b8a84fc55692d3d41fe4d1e5720ff5483a6980808080808080808080a0a319c1c415b271afc0adcb664e67738d103ac168e0bc0b7bd2da7966165cb9518080",
+	}
+
+	rootHash := getCadenceValueArrayFromHexStr(t, "d7f8974fb5ac78d9ac099b9ad5018bedc2ce0a72dad1827a1709da30580f0544")
+
+	addressInHex := "1234567890123456789012345678901234567890"
+
+	address := getCadenceValueArrayFromHexStr(t, addressInHex)
+
+	accountProof := cadence.NewArray([]cadence.Value{
+		// first node encoded
+		getCadenceValueArrayFromHexStr(t, accountProofInHex[0]),
+		// second node encoded
+		getCadenceValueArrayFromHexStr(t, accountProofInHex[1]),
+		// third node encoded
+		getCadenceValueArrayFromHexStr(t, accountProofInHex[2]),
+		// forth node encoded
+		getCadenceValueArrayFromHexStr(t, accountProofInHex[3]),
+	})
+
+	storage := newTestLedger(nil, nil)
+
+	runtimeInterface := &testRuntimeInterface{
+		storage: storage,
+		decodeArgument: func(b []byte, t cadence.Type) (value cadence.Value, err error) {
+			return json.Decode(b)
+		},
+		hash: func(
+			data []byte,
+			tag string,
+			hashAlgorithm HashAlgorithm,
+		) ([]byte, error) {
+			dataInHex := hex.EncodeToString(data)
+			assert.Equal(t, HashAlgorithmKECCAK_256, hashAlgorithm)
+			if dataInHex == accountProofInHex[0] {
+				return hex.DecodeString("d7f8974fb5ac78d9ac099b9ad5018bedc2ce0a72dad1827a1709da30580f0544")
+			}
+			if dataInHex == accountProofInHex[1] {
+				return hex.DecodeString("9287b5c21f2254af4e64fca76acc5cd87399c7f1ede818db4326c98ce2dc2208")
+			}
+			if dataInHex == accountProofInHex[2] {
+				return hex.DecodeString("6163b53d956c55544390c13634ea9aa75309f4fd866f312586942daf0f60fb37")
+			}
+			if dataInHex == accountProofInHex[3] {
+				return hex.DecodeString("41dda0a40a21ade3a20fcd1a4abb2a42b74e9a32b02424ff8db4ea708a5e0fb9")
+			}
+			// hash value for address 1234567890123456789012345678901234567890
+			if dataInHex == addressInHex {
+				return hex.DecodeString("b6979620706f8c652cfb6bf6e923f5156eadd5abaf4022a0b19d52ada089475f")
+			}
+
+			return nil, errors.New("Unknown input to the hash method")
+		},
+		log: func(message string) {
+			fmt.Println(message)
+		},
+	}
+
+	_, err := runtime.ExecuteScript(
+		Script{
+			Source:    script,
+			Arguments: encodeArgs([]cadence.Value{rootHash, address, accountProof}),
+		},
+		Context{
+			Interface: runtimeInterface,
+			Location:  utils.TestLocation,
+		},
+	)
+	require.NoError(t, err)
+}
```

Comment:
Maybe assert the logged messages here:

```suggestion
	require.Equal(t, 
		[]string{"0", "11", "1", "6", "2", "9", "3", "7"}, 
		logMessages,
	)

}
```